### ISSUE TITLE
[Forecasting] Implement XGBoost + LightGBM with walk-forward CV

### ISSUE THREAD

**Issue by alessiopagliarulo** [alessiopagliarulo/walmart-m5-forecasting#4]
Implement gradient-boosted models (XGBoost, LightGBM) with walk-forward cross-validation appropriate for time series (no leakage across the validation folds).
